### PR TITLE
bugfix: Update documentation to fix KeyError in demo code in quickstart

### LINKED ISSUE #10539: [DOC-FIX] Code for the 'Comparing Runs' quickstart throws KeyError
### Willingness to contribute

Yes. I can contribute a documentation fix independently.

### URL(s) with the issue

https://mlflow.org/docs/latest/getting-started/quickstart-2/index.html

### Description of proposal (what needs changing)

The code for logging the final best model in the 'Run a hyperparameter sweep' section has the following line:
```python
    mlflow.tensorflow.log_model(best["model"], "model", signature=signature)
 ````

but the `best` variable returned by `fmin` doesn't include the model, so this throws a KeyError - what you want is `best_run`, which has the loss, status, and model.

### PR DESCRIPTION
**Pull request by wllgrnt**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/wllgrnt/mlflow/pull/10540?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10540/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10540
```

</p>
</details>

### Related Issues/PRs

Resolve #10539 
### What changes are proposed in this pull request?
Updates the documentation so that the demo code  [here](https://mlflow.org/docs/latest/getting-started/quickstart-2/index.html) doesn't throw a KeyError when run.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

image

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [x] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

This one-liner change updates the hyperopt usage example code in the quickstart documentaion so that the correct variable is referenced when logging the best tensorflow model.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by B-Step62 — APPROVED** [review]
@wllgrnt LGTM! Thank you so much for catching this issue and proactively making the fix! Will merge once all CI tests passed.

**Comment by B-Step62** [comment]
fyi: it seems some jobs were stuck, re-opening to trigger them

**Comment by B-Step62** [comment]
@wllgrnt Very weird but some jobs are still not executing. Could you run followings to force triggering them again? While it seems to be an intermittent issue in Github, I can investigate if that happens again. Thanks!

```
$ git commit -sm --amend --no-edit   
$ git push origin HEAD -f
```

**Review by chenmoneygithub — APPROVED** [review]
Thanks for the PR!

**Comment by wllgrnt** [comment]
> @wllgrnt Very weird but some jobs are still not executing. Could you run followings to force triggering them again? While it seems to be an intermittent issue in Github, I can investigate if that happens again. Thanks!
> 
> ```
> $ git commit -sm --amend --no-edit   
> $ git push origin HEAD -f
> ```

The force-push above doesn't seem to have triggered the jobs, let me know if there's something else I should try
